Run 50 pod kubelet resource tracking test on GCE only

yujuhong opened the pull request:
This test launches 50 pods per node, which is over the default --max-pods
setting. GCE test would not have this problem due to the test config override
which sets --max-pod=100. This change runs the 50-pod test only on GCE.

This PR fixes #14620

yujuhong commented:
This PR depends on #14603, which hopefully will land soon.

yujuhong commented:
@dchen1107 @brendandburns @vishh

dchen1107 commented on test/e2e/kubelet_perf.go:
```diff
@@ -73,16 +73,25 @@ var _ = Describe("Kubelet", func() {
+			{podsPerNode: 0, onlyRunOnGCE: false},
+			// 50 is greater the the default --max-pods on kubelet; only run it
+			// on GCE.
+			{podsPerNode: 50, onlyRunOnGCE: true},
```
Why we want to skip GKE here? Shouldn't this is a good test on whatever Jenkin project which is designed to run slow tests?

vishh commented:
Overall LGTM.

yujuhong committed:
e2e: improve sleep time in kubelet resource usage tracking test

yujuhong committed:
Split kubelet e2e resource usage tracking tests

Some tests in this test suite expects --max-pods (i.e. the maximum pod capacity
on kubelet) to be greater than default, which applies only to the GCE test
environment. Split the tests into two sets so that we can better categorize
the tests in the jenkins setup, without making the test itself aware of the
environment.

yujuhong commented:
@dchen1107 

Implications of this PR:
 1. There are now 3 tests: 0, 35, 50 pods per node. "0" and "35" are for benchmarking the resource usage for v1.1, while "50" is for further improving the scalability of kubelet.
 2. GCE builds that run slow tests will run all three tests.
 3. GKE/AWS that run slow tests will run "0" and "35" tests.
 4. I excluded SLOW tests on some GKE builds (e.g., gke-ci) to avoid build timeouts.

This would greatly increase the duration of some builds, so I will keep an eye on jenkins.
We might want to skip SLOW tests on more builds...

dchen1107 commented:
LGTM
